src/plugins/intel_cpu/src/emitters/snippets/aarch64/jit_binary_call_emitter.cpp

```diff
@@ -0,0 +1,126 @@
+#include <cpu/aarch64/cpu_isa_traits.hpp>
+#include <cpu/aarch64/jit_generator.hpp>
+#include <cstddef>
+#include <set>
+#include <utility>
+#include <vector>
+
+#include "emitters/utils.hpp"
+#include "snippets/emitter.hpp"
+#include "utils/general_utils.h"
+
+using namespace dnnl::impl::cpu::aarch64;
+
+namespace ov::intel_cpu::aarch64 {
+
+jit_binary_call_emitter::jit_binary_call_emitter(jit_generator* h, cpu_isa_t isa, std::set<snippets::Reg> live_regs)
+    : jit_emitter(h, isa),
+      m_regs_to_spill(std::move(live_regs)) {}
+
+const std::set<snippets::Reg>& jit_binary_call_emitter::get_regs_to_spill() const {
+    OV_CPU_JIT_EMITTER_ASSERT(m_regs_initialized, "Binary call registers must be initialized first");
+    return m_regs_to_spill;
+}
+
+const Xbyak_aarch64::XReg& jit_binary_call_emitter::get_call_address_reg() const {
+    OV_CPU_JIT_EMITTER_ASSERT(m_regs_initialized, "Binary call registers must be initialized first");
+    return m_call_address_reg;
+}
+
+const Xbyak_aarch64::XReg& jit_binary_call_emitter::get_callee_saved_reg() const {
+    OV_CPU_JIT_EMITTER_ASSERT(m_regs_initialized, "Binary call registers must be initialized first");
+    return m_callee_saved_reg;
+}
+
+void jit_binary_call_emitter::init_binary_call_regs(size_t num_binary_args,
+                                                    const std::vector<size_t>& used_gpr_idxs) const {
+    if (m_regs_initialized) {
+        return;
+    }
+
+    // ARM64 AAPCS: X0-X7 are parameter/result registers, X8-X18 are temporary registers
+    // X19-X28 are callee-saved registers, X29 is frame pointer, X30 is link register, X31 is stack pointer
+
+    std::vector<size_t> all_used_idxs = used_gpr_idxs;
+
+    // Add ABI parameter registers to used list (X0-X7 based on num_binary_args)
+    for (size_t i = 0; i < std::min(num_binary_args, static_cast<size_t>(8)); i++) {
+        all_used_idxs.push_back(i);
+        m_regs_to_spill.emplace(snippets::RegType::gpr, i);
+    }
+
+    // Add special registers that should not be allocated
+    std::vector<size_t> reserved_regs = {
+        18,  // Platform register (should not be used)
+        29,  // Frame pointer (FP)
+        30,  // Link register (LR)
+        31   // Stack pointer (SP)
+    };
+    all_used_idxs.insert(all_used_idxs.end(), reserved_regs.begin(), reserved_regs.end());
+
+    // Allocate call address register from temporary registers (X8-X18, excluding X18)
+    std::vector<size_t> temp_candidates;
+    for (size_t i = 8; i < 18; i++) {
+        if (std::find(all_used_idxs.begin(), all_used_idxs.end(), i) == all_used_idxs.end()) {
+            temp_candidates.push_back(i);
+        }
+    }
+
+    OV_CPU_JIT_EMITTER_ASSERT(!temp_candidates.empty(), "No available temporary register for call address");
+    m_call_address_reg = Xbyak_aarch64::XReg(static_cast<int>(temp_candidates[0]));
+    all_used_idxs.push_back(temp_candidates[0]);
+    m_regs_to_spill.emplace(snippets::RegType::gpr, temp_candidates[0]);
```

Comment:
Do we really need to allocate entire vector to find only one register?
The same question for `m_callee_saved_reg`